--- usr.sbin/sysinstall/anonFTP.c.orig	2010-08-24 04:54:18.000000000 -0700
+++ usr.sbin/sysinstall/anonFTP.c	2010-08-24 05:16:10.000000000 -0700
@@ -178,7 +178,7 @@ createFtpUser(void)
     fprintf(fptr, "%s", pwline);
     fclose(fptr);
     msgNotify("Remaking password file: %s", _PATH_MASTERPASSWD);
-    vsystem("pwd_mkdb -p %s", _PATH_MASTERPASSWD);
+    vsystem("%spwd_mkdb -p %s", xarchInstall() ? "/stand/" : "", _PATH_MASTERPASSWD);
     return DITEM_SUCCESS | DITEM_RESTORE;
 }
 
@@ -282,25 +282,25 @@ configAnonFTP(dialogMenuItem *self __unu
     /*** If HomeDir does not exist, create it ***/
     
     if (!directory_exists(tconf.homedir))
-	vsystem("mkdir -p %s", tconf.homedir);
+	vsystem("%smkdir -p %s", xarchInstall() ? "/stand/" : "", tconf.homedir);
     
     if (directory_exists(tconf.homedir)) {
 	msgNotify("Configuring %s for use by anon FTP.", tconf.homedir);
-	vsystem("chmod 555 %s && chown root:%s %s", tconf.homedir, tconf.group, tconf.homedir);
-	vsystem("mkdir %s/etc && chmod 555 %s/etc", tconf.homedir, tconf.homedir);
-	vsystem("mkdir -p %s/pub", tconf.homedir);
+	vsystem("%schmod 555 %s && %schown root:%s %s", xarchInstall() ? "/stand/" : "", tconf.homedir, xarchInstall() ? "/stand/" : "", tconf.group, tconf.homedir);
+	vsystem("%smkdir %s/etc && %schmod 555 %s/etc", xarchInstall() ? "/stand/" : "", tconf.homedir, xarchInstall() ? "/stand/" : "", tconf.homedir);
+	vsystem("%smkdir -p %s/pub", xarchInstall() ? "/stand/" : "", tconf.homedir);
 	if (tconf.upload[0]) {
-	    vsystem("mkdir -p %s/%s", tconf.homedir, tconf.upload);
-	    vsystem("chmod 1777 %s/%s", tconf.homedir, tconf.upload);
+	    vsystem("%smkdir -p %s/%s", xarchInstall() ? "/stand/" : "", tconf.homedir, tconf.upload);
+	    vsystem("%schmod 1777 %s/%s", xarchInstall() ? "/stand/" : "", tconf.homedir, tconf.upload);
 	}
 	
 	if (DITEM_STATUS(createFtpUser()) == DITEM_SUCCESS) {
 	    msgNotify("Copying password information for anon FTP.");
-	    vsystem("awk -F: '{if ((substr($1, 1, 1) != \"+\") && (substr($1, 1, 1) != \"-\") && ($3 < 10 || $1 == \"ftp\")) print $0}' /etc/master.passwd > %s/etc/master.passwd", tconf.homedir);
-	    vsystem("/usr/sbin/pwd_mkdb -d %s/etc %s/etc/master.passwd && chmod 444 %s/etc/pwd.db", tconf.homedir, tconf.homedir, tconf.homedir);
-	    vsystem("rm -f %s/etc/master.passwd %s/etc/spwd.db", tconf.homedir, tconf.homedir);
-	    vsystem("awk -F: '!/^#/ {if ((substr($1, 1, 1) != \"+\") && (substr($1, 1, 1) != \"-\") && ($3 < 100)) printf \"%%s:*:%%s:\\n\", $1, $3}' /etc/group > %s/etc/group && chmod 444 %s/etc/group", tconf.homedir, tconf.homedir);
-	    vsystem("chown -R root:%s %s/pub", tconf.group, tconf.homedir);
+	    vsystem("%sawk -F: '{if ((substr($1, 1, 1) != \"+\") && (substr($1, 1, 1) != \"-\") && ($3 < 10 || $1 == \"ftp\")) print $0}' /etc/master.passwd > %s/etc/master.passwd", xarchInstall() ? "/stand/" : "", tconf.homedir);
+	    vsystem("%s/pwd_mkdb -d %s/etc %s/etc/master.passwd && %schmod 444 %s/etc/pwd.db", xarchInstall() ? "/stand" : "/usr/sbin", tconf.homedir, tconf.homedir, xarchInstall() ? "/stand/" : "", tconf.homedir);
+	    vsystem("%srm -f %s/etc/master.passwd %s/etc/spwd.db", xarchInstall() ? "/stand/" : "", tconf.homedir, tconf.homedir);
+	    vsystem("%sawk -F: '!/^#/ {if ((substr($1, 1, 1) != \"+\") && (substr($1, 1, 1) != \"-\") && ($3 < 100)) printf \"%%s:*:%%s:\\n\", $1, $3}' /etc/group > %s/etc/group && %schmod 444 %s/etc/group", xarchInstall() ? "/stand/" : "", tconf.homedir, xarchInstall() ? "/stand/" : "", tconf.homedir);
+	    vsystem("%schown -R root:%s %s/pub", xarchInstall() ? "/stand/" : "", tconf.group, tconf.homedir);
 	}
 	else {
 	    msgConfirm("Unable to create FTP user!  Anonymous FTP setup failed.");
