--- usr.sbin/sysinstall/tcpip.c.orig	2010-08-24 07:18:30.000000000 -0700
+++ usr.sbin/sysinstall/tcpip.c	2010-08-24 07:20:14.000000000 -0700
@@ -394,13 +394,13 @@ tcpOpenDialog(Device *devp)
 	    sysctlbyname("net.inet6.ip6.forwarding", NULL, 0, &i, sizeof(i));
 	    i = 1;
 	    sysctlbyname("net.inet6.ip6.accept_rtadv", NULL, 0, &i, sizeof(i));
-	    vsystem("ifconfig %s up", devp->name);
+	    vsystem("%sifconfig %s up", xarchInstall() ? "/stand/" : "", devp->name);
 	    len = sizeof(i);
 	    sysctlbyname("net.inet6.ip6.dad_count", &i, &len, NULL, 0);
 	    sleep(i + 1);
 	    Mkdir("/var/run");
 	    msgNotify("Scanning for RA servers...");
-	    if (0 == vsystem("rtsol %s", devp->name)) {
+	    if (0 == vsystem("%srtsol %s", xarchInstall() ? "/stand/" : "", devp->name)) {
 		len = sizeof(i);
 		sysctlbyname("net.inet6.ip6.dad_count", &i, &len, NULL, 0);
 		sleep(i + 1);
@@ -425,7 +425,7 @@ tcpOpenDialog(Device *devp)
 	    msgNotify("Scanning for DHCP servers...");
 	    /* XXX clear any existing lease */
 	    /* XXX limit protocol to N tries */
-	    if (0 == vsystem("dhclient %s", devp->name)) {
+	    if (0 == vsystem("%sdhclient %s", xarchInstall() ? "/stand/" : "", devp->name)) {
 		dhcpGetInfo(devp);
 		use_dhcp = TRUE;
 	    }
